projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm.e2e-spec.ts

```diff
@@ -0,0 +1,239 @@
+      agentSignOut();
+
+      cy.get('button[title="Close ASM"]').click();
+      cy.get('cx-asm-main-ui').should('exist');
+      cy.get('cx-asm-main-ui').should('not.be.visible');
+    });
+  });
+
+  describe('Customer Self Verification', () => {
+    it('checks data changes made by the agent', () => {
+      cy.log('--> customer sign in');
+      cy.visit('/login');
+      loginCustomerInStorefront();
+      assertCustomerIsSignedIn();
+
+      cy.log('Check personal details updated by the agent');
+      cy.selectUserMenuOption({
+        option: 'Personal Details',
+      });
+      profile.verifyUpdatedProfile();
+
+      cy.log('--> check address created by the agent');
+      cy.selectUserMenuOption({
+        option: 'Address Book',
+      });
+      cy.get('cx-card').should('have.length', 1);
+      addressBook.verifyNewAddress();
+
+      cy.log('--> Check consent given by agent');
+      cy.selectUserMenuOption({
+        option: 'Consent Management',
+      });
+      cy.get('input[type="checkbox"]').first().should('be.checked');
+
+      checkout.signOutUser();
+    });
+  });
+
+  describe('When a customer session and an asm agent session are both active', () => {
+    it('Customer should not be able to login when there is an active CS agent session.', () => {
+      const loginPage = checkout.waitForPage('/login', 'getLoginPage');
+      cy.visit('/login?asm=true');
+      cy.wait(`@${loginPage}`);
+
+      agentLogin();
+      login(customer.email, customer.password);
+      getErrorAlert().should(
+        'contain',
+        'Cannot login as user when there is an active CS agent session. Please either emulate user or logout CS agent.'
+      );
+    });
+
+    // TODO(#9445): Add e2e test for this scenario
+    it.skip('agent login when user is logged in should start this user emulation', () => {});
+
+    // TODO(#9445): Add e2e test for this scenario
+    it.skip('agent logout when user was logged and emulated should restore the session', () => {});
+  });
+});
+
+function listenForAuthenticationRequest(): string {
+  return interceptPost(
+    'csAgentAuthentication',
+    '/authorizationserver/oauth/token'
+  );
+}
+
+export function listenForCustomerSearchRequest(): string {
+  return interceptGet(
+    'customerSearch',
+    '/assistedservicewebservices/customers/search?*'
+  );
+}
+
+function listenForUserDetailsRequest(): string {
+  return interceptGet('userDetails', '/users/*', true);
+}
+
+export function agentLogin(): void {
+  const authRequest = listenForAuthenticationRequest();
+
+  cy.get('cx-storefront').within(() => {
+    cy.get('cx-csagent-login-form').should('exist');
+    cy.get('cx-customer-selection').should('not.exist');
+    cy.get('cx-csagent-login-form form').within(() => {
+      cy.get('[formcontrolname="userId"]').type('asagent');
+      cy.get('[formcontrolname="password"]').type('pw4all');
+      cy.get('button[type="submit"]').click();
+    });
+  });
+
+  cy.wait(authRequest);
```

Comment:
Why not verify the response code 200 anymore when we wait?

Reply:
It's implicit. If you don't get a 200, the test will fail. Unless you get an equivalent code (20x for example), where your response will still be valid. We should only validate status codes when it's relevant or important to the test